nixos/doc/manual/development/settings-options.xml

```diff
@@ -0,0 +1,169 @@
+<section xmlns="http://docbook.org/ns/docbook"
+        xmlns:xlink="http://www.w3.org/1999/xlink"
+        xmlns:xi="http://www.w3.org/2001/XInclude"
+        version="5.0"
+        xml:id="sec-settings-options">
+ <title>Options for Program Settings</title>
+
+ <para>
+   Many programs have configuration files where program-specific settings can be declared. File formats can be separated into two categories:
+   <itemizedlist>
+     <listitem>
+       <para>
+         Nix-representable ones: These can trivially be mapped to a subset of Nix syntax. E.g. JSON is an example, since its values like <literal>{"foo":{"bar":10}}</literal> can be mapped directly to Nix: <literal>{ foo = { bar = 10; }; }</literal>. Other examples are INI, YAML and TOML. The following section explains the convention for these settings.
```

Comment:
It's really annoying that our xml formatter does not break lines.
This is very hard to read on github.

Reply:
Oh good point. I could introduce manual line breaks, but those get annoying quickly when the text needs changes.

Reply:
This is [intentional](https://github.com/NixOS/nixpkgs/pull/68181) since hard wrapping breaks diffs. On the other hand, it is easy to enable soft wrapping using an user style. I recommend https://github.com/StylishThemes/GitHub-code-wrap.

Reply:
I see. The only alternative that would make both github code review and diffs happy is to use
break on every sentence instead of each paragraph. No idea if our xmlformatter even would have an option for that.

Reply:
Given that we're months away switching to readable documentation I wouldn't bother right now.